lib/hooks/useCoingeckoTokenData.js

```diff
@@ -0,0 +1,35 @@
+import axios from 'axios'
+import { useQuery } from 'react-query'
+
+import { useNetwork } from 'lib/hooks/useNetwork'
+import Dai from 'assets/images/dai.svg'
+
+const COIN_GECKO_API_URL = 'https://api.coingecko.com/api/v3'
+
+export const useCoingeckoTokenData = (contractAddress) => {
+  const { id: networkId } = useNetwork()
+
+  return useQuery([contractAddress, networkId], () => getgoingeckoTokenData(contractAddress), {
```

Comment:
This needs some sort of string qualifier. The key should end up being similar to a react component key (ie. `coingecko-token-data-${networkId}-${tokenAddress}`